client/components/forms/form-label/index.js

```diff
@@ -0,0 +1,48 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import { localize } from 'i18n-calypso';
+import { isObject, omit } from 'lodash';
+
+const renderRequiredBadge = translate => (
+	<small className="form-label__required">{ translate( 'Required' ) }</small>
+);
+
+const renderOptionalBadge = translate => (
+	<small className="form-label__optional">{ translate( 'Optional' ) }</small>
+);
+
+const addKeys = elements =>
+	elements.map( ( elem, idx ) => ( isObject( elem ) ? { ...elem, key: idx } : elem ) );
+
+const FormLabel = ( { children, required, optional, translate, className, ...extraProps } ) => {
+	children = React.Children.toArray( children ) || [];
+	if ( required ) {
+		children.push( renderRequiredBadge( translate ) );
+	}
+
+	if ( optional ) {
+		children.push( renderOptionalBadge( translate ) );
+	}
+
+	return (
+		<label
+			{ ...omit( extraProps, 'moment', 'numberFormat' ) }
+			className={ classnames( className, 'form-label' ) }
+		>
+			{ children.length ? addKeys( children ) : null }
```

Comment:
I'd recommend doing this, which prevents from doing `addKeys` after cloning the elements, overwriting the `children` through `React.Children.toArray`, as well as the `lodash` dependency :

```
diff --git a/client/components/forms/form-label/index.js b/client/components/forms/form-label/index.js
index 81b6f425..58ff79e3 100644
--- a/client/components/forms/form-label/index.js
+++ b/client/components/forms/form-label/index.js
@@ -8,41 +8,24 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import classnames from 'classnames';
 import { localize } from 'i18n-calypso';
-import { isObject, omit } from 'lodash';
-
-const renderRequiredBadge = translate => (
-	<small className="form-label__required">{ translate( 'Required' ) }</small>
-);
-
-const renderOptionalBadge = translate => (
-	<small className="form-label__optional">{ translate( 'Optional' ) }</small>
-);
-
-const addKeys = elements =>
-	elements.map( ( elem, idx ) => ( isObject( elem ) ? { ...elem, key: idx } : elem ) );
-
-const FormLabel = ( { children, required, optional, translate, className, ...extraProps } ) => {
-	children = React.Children.toArray( children ) || [];
-	if ( required ) {
-		children.push( renderRequiredBadge( translate ) );
-	}
-
-	if ( optional ) {
-		children.push( renderOptionalBadge( translate ) );
-	}
 
+const FormLabel = ( { children, required, optional, translate, className, moment, numberFormat, ...extraProps } ) => {
 	return (
 		<label
-			{ ...omit( extraProps, 'moment', 'numberFormat' ) }
+			{ ...extraProps }
 			className={ classnames( className, 'form-label' ) }
 		>
-			{ children.length ? addKeys( children ) : null }
+			{children}
+			{required && ( <small className="form-label__required">{ translate( 'Required' ) }</small> ) }
+			{optional && ( <small className="form-label__optional">{ translate( 'Optional' ) }</small> )}
 		</label>
 	);
 };
 
 FormLabel.propTypes = {
 	required: PropTypes.bool,
+	optional: PropTypes.bool,
+	children: PropTypes.node,
 };
 
 export default localize( FormLabel );
```

(pardon the formatting)

Reply:
Wow that is a lot cleaner, thanks for taking a look at this!! I've pushed these changes up now. The only difference I saw was with the key on the children, but it still worked as expected.